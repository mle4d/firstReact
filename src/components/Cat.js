import React from 'react';

export default function Cat({ name, age, weight }) {
  const cat = {
    name,
    age,
    weight
  };
  const fields = Object.keys(cat)
    .map(key => {
      return (
        <>
          <dt>{key}</dt>
          <dd>{cat[key]}</dd>
        </>
      );
    });
  return (
    <dl>
      {fields}
    </dl>
  );
}
