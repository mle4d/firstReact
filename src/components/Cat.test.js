import React from 'react';
import { shallow } from 'enzyme';
import Cat from './Cat';

describe('App component', () => {
  it('renders App', () => {
    const wrapper = shallow(<Cat />);
    expect(wrapper).toMatchSnapshot();
  });
});
