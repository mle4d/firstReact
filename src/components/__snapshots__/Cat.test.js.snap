// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<dl>
  <dt>
    name
  </dt>
  <dd>
    Darla
  </dd>
  <dt>
    age
  </dt>
  <dd>
    5
  </dd>
  <dt>
    weight
  </dt>
  <dd>
    8 lbs
  </dd>
</dl>
`;
